xonsh/built_ins.py

```diff
@@ -555,6 +564,16 @@ def run_subproc(cmds, captured=True):
                 except PermissionError:
                     e = 'xonsh: subprocess mode: permission denied: {0}'
                     raise XonshError(e.format(cmd[0]))
+        _stdin_file = None
+        if captured == 'object' and stdin is not None:
+            _stdin_file = tempfile.NamedTemporaryFile()
+            cproc = Popen(['cat'],
+                          stdin=stdin,
+                          stdout=PIPE)
+            tproc = Popen(['tee', _stdin_file.name],
```

Comment:
I am not afraid of capturing stdin here, since we are never capturing it interactively (only when it comes from a pipe or a file redirect).  I thought it might be nice, since we're capturing stdout and stderr anyway, just to grab all three.  Maybe someone will have a use for it?

At the least, I will wrap it with a check for whether `tee` and `cat` exist.  I can also add the environment variable if you feel strongly enough, but it isn't something that worries me personally.  I guess it could be seen a security vulnerability...